entry/src/main/java/com/thoughtbot/expandablerecyclerview/entities/CommonHelper.java

```diff
@@ -0,0 +1,253 @@
+        mGroupNameItem.add(new ParentChild(null, ResUtil.getString(context, ResourceTable.String_item_Jazz)));
+        mGroupNameItem.add(new ParentChild(null, ResUtil.getString(context, ResourceTable.String_item_Classic)));
+        mGroupNameItem.add(new ParentChild(null, ResUtil.getString(context, ResourceTable.String_item_Salsa)));
+        mGroupNameItem.add(new ParentChild(null, ResUtil.getString(context, ResourceTable.String_item_Bluegrass)));
+        mFinalGroupNameItem.add(ResUtil.getString(context, ResourceTable.String_item_Rock));
+        mFinalGroupNameItem.add(ResUtil.getString(context, ResourceTable.String_item_Jazz));
+        mFinalGroupNameItem.add(ResUtil.getString(context, ResourceTable.String_item_Classic));
+        mFinalGroupNameItem.add(ResUtil.getString(context, ResourceTable.String_item_Salsa));
+        mFinalGroupNameItem.add(ResUtil.getString(context, ResourceTable.String_item_Bluegrass));
+    }
+
+    /**
+     * This method is used to prepare group items images.
+     */
+    public void getGroupIcons() {
+        mGroupImageItem.add(ResourceTable.Media_rock);
+        mGroupImageItem.add(ResourceTable.Media_jazz);
+        mGroupImageItem.add(ResourceTable.Media_classic);
+        mGroupImageItem.add(ResourceTable.Media_salsa);
+        mGroupImageItem.add(ResourceTable.Media_bluegrass);
+    }
+
+    /**
+     * This method is used to prepare Favourite child items.
+     */
+    public void getFavouriteItems() {
+        mFavouriteItem.add(ResUtil.getString(context, ResourceTable.String_item_child_Rock1));
+        mFavouriteItem.add(ResUtil.getString(context, ResourceTable.String_item_child_Rock4));
+        mFavouriteItem.add(ResUtil.getString(context, ResourceTable.String_item_child_Jazz1));
+        mFavouriteItem.add(ResUtil.getString(context, ResourceTable.String_item_child_Jazz2));
+        mFavouriteItem.add(ResUtil.getString(context, ResourceTable.String_item_child_Classic2));
+        mFavouriteItem.add(ResUtil.getString(context, ResourceTable.String_item_child_Salsa1));
+        mFavouriteItem.add(ResUtil.getString(context, ResourceTable.String_item_child_Bluegrass1));
+    }
+
+    /**
+     * for handling both parent and child item.
+     *
+     * @param holder holding the view.
+     * @param text group/child text item to be displayed.
+     * @param image group image to be displayed.
+     */
+    public void handleListItem(ExpandableListAdapter.ViewHolder holder, ParentChild text, Integer image) {
+        if (!mTempChildNameItem.contains(text.getChildItem())) {
+            handleParentItem(holder, text, image);
+        } else {
+            handleChildItem(holder, text);
+        }
+    }
+
+    /**
+     * for setting the adapter and OnItemClickListener.
+     *
+     * @param expandableListAdapter adpter for holding the list data.
+     */
+    public void setterfunction(ExpandableListAdapter<ParentChild> expandableListAdapter) {
```

Comment:
method name must be meaningful and break the method for setting listener and adapter